diff --git a/hadoop/conf/local/hive-site.xml b/hadoop/conf/local/hive-site.xml
index 797c3dc..d4ad777 100644
--- a/hadoop/conf/local/hive-site.xml
+++ b/hadoop/conf/local/hive-site.xml
@@ -395,7 +395,7 @@
 
 <property>
   <name>hive.mr3.aux.jars</name>
-  <value>hive-llap-common-4.0.0.jar,hive-llap-server-4.0.0.jar,hive-llap-tez-4.0.0.jar,hive-iceberg-handler-4.0.0.jar,log4j-1.2-api-2.18.0.jar,orc-core-2.0.3.jar,orc-format-1.0.0.jar,orc-shims-2.0.3.jar,orc-tools-2.0.3.jar</value>
+  <value>hive-llap-common-3.1.3.jar,hive-llap-server-3.1.3.jar,hive-llap-tez-3.1.3.jar</value>
 </property>
 
 <property>
diff --git a/hadoop/conf/tpcds/hive-site.xml b/hadoop/conf/tpcds/hive-site.xml
index 36e50ca..755669a 100644
--- a/hadoop/conf/tpcds/hive-site.xml
+++ b/hadoop/conf/tpcds/hive-site.xml
@@ -510,11 +510,7 @@
     
 <property>
   <name>hive.mr3.aux.jars</name>
-  <value>hive-llap-common-4.0.0.jar,hive-llap-server-4.0.0.jar,hive-llap-tez-4.0.0.jar,hive-iceberg-handler-4.0.0.jar,log4j-1.2-api-2.18.0.jar,orc-core-2.0.3.jar,orc-format-1.0.0.jar,orc-shims-2.0.3.jar,orc-tools-2.0.3.jar</value>
-  <description>
-    Include Iceberg jar, even if mr3.am.permit.custom.user.class is set to true,
-    because IcebergInputFormat may use URLClassLoader.
-  </description>
+  <value>hive-llap-common-3.1.3.jar,hive-llap-server-3.1.3.jar,hive-llap-tez-3.1.3.jar</value>
 </property>
 
 <property>
diff --git a/hadoop/env.sh b/hadoop/env.sh
index d6a6a0b..64ca3ed 100755
--- a/hadoop/env.sh
+++ b/hadoop/env.sh
@@ -201,7 +201,7 @@ MR3_REV=2.0
 TEZ_REV=0.9.1.mr3.${MR3_REV}
 
 # Hive-MR3 revision number
-HIVE_REV=4.0.0
+HIVE_REV=3.1.3
 
 #
 # Step 10. High availability (optional)
diff --git a/helm/hive/values.yaml b/helm/hive/values.yaml
index f444d3e..2447d93 100644
--- a/helm/hive/values.yaml
+++ b/helm/hive/values.yaml
@@ -25,8 +25,8 @@ name:
   persistentVolumeClaim: workdir-pvc
 
 docker:
-  image: mr3project/hive:4.0.0.mr3.2.2
-  containerWorkerImage: mr3project/hive:4.0.0.mr3.2.2
+  image: mr3project/hive:3.1.3.mr3.2.2
+  containerWorkerImage: mr3project/hive:3.1.3.mr3.2.2
   # do not use a composite name like hive@RED, hive/red0@RED
   user: hive
   imagePullPolicy: IfNotPresent
diff --git a/kubernetes/env.sh b/kubernetes/env.sh
index 241474b..f62b853 100755
--- a/kubernetes/env.sh
+++ b/kubernetes/env.sh
@@ -51,7 +51,7 @@ CREATE_SERVICE_ACCOUNTS=true
 # Step 1. Docker images
 #
 
-DOCKER_HIVE_IMG=${DOCKER_HIVE_IMG:-mr3project/hive:4.0.0.mr3.2.2}
+DOCKER_HIVE_IMG=${DOCKER_HIVE_IMG:-mr3project/hive:3.1.3.mr3.2.2}
 DOCKER_HIVE_WORKER_IMG=$DOCKER_HIVE_IMG
 
 # do not use a composite name like hive@RED, hive/red0@RED (which results in NPE in ContainerWorker)
diff --git a/kubernetes/yaml/hive.yaml b/kubernetes/yaml/hive.yaml
index 9a53a17..57b0c93 100644
--- a/kubernetes/yaml/hive.yaml
+++ b/kubernetes/yaml/hive.yaml
@@ -34,7 +34,7 @@ spec:
                 matchLabels:
                   hivemr3_app: metastore
       containers:
-      - image: mr3project/hive:4.0.0.mr3.2.2
+      - image: mr3project/hive:3.1.3.mr3.2.2
         command: ["/opt/mr3-run/hive/hiveserver2-service.sh"]
         args: ["start", "--kubernetes"]
         imagePullPolicy: Always
diff --git a/kubernetes/yaml/metastore.yaml b/kubernetes/yaml/metastore.yaml
index 0c8554c..4ab4806 100644
--- a/kubernetes/yaml/metastore.yaml
+++ b/kubernetes/yaml/metastore.yaml
@@ -43,7 +43,7 @@ spec:
         hostnames:
         - "red0"
       containers:
-      - image: mr3project/hive:4.0.0.mr3.2.2
+      - image: mr3project/hive:3.1.3.mr3.2.2
         command: ["/opt/mr3-run/hive/metastore-service.sh"]
         args: ["start", "--kubernetes"]
         imagePullPolicy: Always
